device-api/src/perf_regs.rs

```diff
@@ -136,49 +136,95 @@ impl PerformanceCounter {
         Ok(counter)
     }
 
-    /// Calculate utilization using performance counters at two points in time
-    pub fn calculate_performance_values(&self, other: &Self) -> Option<PerformanceValues> {
+    pub fn cycle_count(&self) -> usize {
+        self.cycle_count
+    }
+
+    pub fn task_execution_cycle(&self) -> u32 {
+        self.task_execution_cycle
+    }
+
+    pub fn tensor_execution_cycle(&self) -> u32 {
+        self.tensor_execution_cycle
+    }
+
+    pub fn calculate_increased(&self, other: &Self) -> Self {
         let (prev, next) = if self.now < other.now {
             (self, other)
         } else {
             (other, self)
         };
 
-        // when the cycle count is reversed, it is considered invalid.
+        // when the cycle count is reversed, NPU was restarted.
         if next.cycle_count < prev.cycle_count {
-            return None;
+            return *next;
         }
 
+        let elapsed_time = next.now.duration_since(prev.now);
+        if elapsed_time.is_err() {
+            return *next;
+        }
+
+        let elapsed_time = elapsed_time.unwrap().as_nanos() as usize;
         let cycle_gap = next.cycle_count - prev.cycle_count;
+
+        // Unfortunately, it's a naive implementation for now.
+        if (cycle_gap as f64 / elapsed_time as f64) < 0.45 {
+            return *next;
+        }
+
         let task_cycle_gap =
             Self::safe_u32_subtract(next.task_execution_cycle, prev.task_execution_cycle);
         let tensor_cycle_gap =
             Self::safe_u32_subtract(next.tensor_execution_cycle, prev.tensor_execution_cycle);
 
-        // If the task cycle is bigger than the total cycle, it is considered invalid.
-        if cycle_gap < task_cycle_gap {
-            return None;
+        let task_cycle_gap = if cycle_gap < task_cycle_gap {
+            cycle_gap
+        } else {
+            task_cycle_gap
+        };
+
+        let tensor_cycle_gap = if task_cycle_gap < tensor_cycle_gap {
+            task_cycle_gap
+        } else {
+            tensor_cycle_gap
+        };
+
+        Self {
+            now: next.now,
+            cycle_count: cycle_gap,
+            task_execution_cycle: if task_cycle_gap > u32::MAX as usize {
+                u32::MAX
+            } else {
+                task_cycle_gap as u32
+            },
```

Comment:
여기도 std::cmp를 쓰면 좋을 것 같습니다.
```suggestion
            task_execution_cycle: std::cmp::min(task_cycle_gap, u32::MAX as usize) as u32,
```